orttraining/orttraining/python/training/ortmodule/_fallback.py

```diff
@@ -180,6 +181,8 @@ def fallback(self, log_level: _logger.LogLevel, *inputs, **kwargs):
                 ),
                 UserWarning,
             )
+            if type(self._exception) == ORTModuleONNXModelException:
```

Comment:
Also if the retry is set, we should probably reset the raised exception flag so we can re-raise the warning.